qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/OverviewView.groovy

```diff
@@ -0,0 +1,155 @@
+    OverviewView(OfferOverviewModel model) {
+        this.model = model
+        this.overviewGrid = new Grid<>()
+        this.downloadBtn = new Button(VaadinIcons.DOWNLOAD)
+        this.downloadSpinner = new ProgressBar()
+        initLayout()
+        setupDataProvider()
+        setupGrid()
+        setupListeners()
+    }
+
+    private void initLayout() {
+        /*
+        We start with the header, that contains a descriptive
+        title of what the view is about.
+         */
+        final def headerRow = new HorizontalLayout()
+        final def label = new Label("Available Offers")
+        label.addStyleName(ValoTheme.LABEL_HUGE)
+        headerRow.addComponent(label)
+        this.addComponent(headerRow)
+
+        /*
+        Below the header, we create content row with two components.
+        The left component will be the offer overview, the
+        right component will be the offer download button.
+         */
+        final def overviewRow = new HorizontalLayout()
+        final def activityContainer = new VerticalLayout()
+        downloadBtn.setStyleName(ValoTheme.BUTTON_LARGE)
+        downloadBtn.setEnabled(false)
+        // Makes the progress bar a spinner
+        downloadSpinner.setIndeterminate(true)
+        downloadSpinner.setVisible(false)
+        activityContainer.addComponents(downloadBtn, downloadSpinner)
+        activityContainer.setMargin(false)
+        overviewRow.addComponents(overviewGrid, activityContainer)
+        this.addComponent(overviewRow)
+
+        overviewRow.setWidthFull()
+        this.setWidthFull()
+    }
+
+    private void setupDataProvider() {
+        overviewGrid.setItems(model.offerOverviewList)
+    }
+
+    private void setupGrid() {
+        def dateColumn = overviewGrid.addColumn({ overview -> overview.getModificationDate() })
+                .setCaption("Date")
+        overviewGrid.addColumn({overview -> overview.getProjectTitle()}).setCaption("Title")
+        overviewGrid.addColumn({overview -> overview.getCustomer()}).setCaption("Customer")
+        overviewGrid.addColumn({overview -> overview.getTotalPrice()}).setCaption("Total Price")
+        overviewGrid.sort(dateColumn, SortDirection.DESCENDING)
+        overviewGrid.setWidthFull()
+    }
+
+    private void setupListeners() {
+        overviewGrid.addSelectionListener(
+                { selection ->
+                    selection.firstSelectedItem.ifPresentOrElse({overview ->
+                        UI.getCurrent().setPollInterval(50)
+                        downloadSpinner.setVisible(true)
+                        new LoadOfferInfoThread(UI.getCurrent(), overview).start()
+                    }, {})
+                })
+    }
+
+    private void createResourceForDownload() {
+        removeExistingResources()
+        StreamResource offerResource =
+                new StreamResource((StreamResource.StreamSource res) -> {
+                    return model.getOfferAsPdf()
+                }, "myoffer.pdf")
```

Comment:
NIT: I think it would make sense to replace the "myoffer" dummy text with a string of the selected offerid down the line

Reply:
when I implemented this loc, there was no such thing as an OfferId implemented :) So i would delay this to another PR, where the ID implementation is also done.